pkg/operator/gcwatchercontroller/gcwatcher_controller.go

```diff
@@ -102,16 +111,24 @@ func (c *GarbageCollectorWatcherController) syncWorker(ctx context.Context, sync
 	}
 	requiredAlertsSet := sets.NewString(c.alertNames...)
 
-	prometheusClient, err := newPrometheusClient(ctx, c.configMapClient)
-	if err != nil && errors.IsNotFound(err) {
-		// Prometheus client when failed to instantiate should not result in error being generated. This makes sure we don't
-		// generate excessive events.
-		return nil
-	} else if err != nil {
-		return err
+	if c.prometheusClient == nil {
+		// Don't create prometheus client on every sync, re-use from struct, this will also
```

Comment:
we have to instantiate the client every time to detect the CA change

alternatively we could detect that the config map changed and reinitialize the client only in that case